src/services/formatting/smartIndenter.ts

```diff
@@ -531,9 +531,20 @@ namespace ts.formatting {
             return false;
         }
 
-        export function nodeWillIndentChild(parent: TextRangeWithKind, child: TextRangeWithKind | undefined, indentByDefault: boolean): boolean {
+        export function nodeWillIndentChild(parent: TextRangeWithKind, child: TextRangeWithKind | undefined, sourceFile: SourceFileLike | undefined, indentByDefault: boolean): boolean {
             const childKind = child ? child.kind : SyntaxKind.Unknown;
+
             switch (parent.kind) {
+                case SyntaxKind.VariableDeclaration:
+                case SyntaxKind.PropertyAssignment:
+                case SyntaxKind.ObjectLiteralExpression:
+                    if (sourceFile && childKind === SyntaxKind.ObjectLiteralExpression) {
+                        const childStart = skipTrivia(sourceFile.text, child.pos);
+                        const startLine = sourceFile.getLineAndCharacterOfPosition(childStart).line;

```

Comment:
My comment's marked outdated since this code was moved: This could use an `isSingleLineNode(Node, SourceFile)` helper function.